app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -14,15 +14,28 @@ def sync_monitor
   end
 
   def sync_from_system
+    # All miq_server instances have to reside on the same Kubernetes cluster, so
+    # we only have to sync the list of pods and deployments once
+    return unless my_server.is_master?
+
     ensure_kube_monitors_started
-    cleanup_orphaned_worker_rows
-    sync_deployment_settings
   end
 
   def enough_resource_to_start_worker?(_worker_class)
     true
   end
 
+  def sync_workers
+    # Before syncing the workers check for any orphaned worker rows that don't have
+    # a current pod and delete them
+    cleanup_orphaned_worker_rows
```

Comment:
Wondering if this belongs in the `cleanup_failed_workers` section

Reply:
We may have to ensure that the master server goes first in the list...otherwise if a non-master server goes first it will try to compare to an empty "global" list.

Reply:
Either that or we spawn off the kubernetes watches as a completely separate thing beforehand, and then piggy back on the master server only to ensure the thread is still alive ornot.

Reply:
It's unfortunate podified does this thing where it creates servers for zones.  Anything that hits the miq_workers table is scoped for the server of the zone that it's currently processing, but anything that looks at current_pods is global and needs the check to make sure only the primary server does it.  

It does make sense to to put this with `delete_failed_deployments` in `cleanup_failed_workers` as long as the timing works but the latter is global so it has the `primary?` check whereas this is for each server and doesn't.  It's similar behavior at a different scope.